data-prepper-pipeline-parser/src/test/java/org/opensearch/dataprepper/pipeline/parser/transformer/DynamicConfigTransformerTest.java

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.opensearch.dataprepper.pipeline.parser.transformer;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+import com.fasterxml.jackson.dataformat.yaml.YAMLGenerator;
+import com.jayway.jsonpath.PathNotFoundException;
+import static org.assertj.core.api.Assertions.assertThat;
```

Comment:
Please use Hamcrest. Then we can remove the assertj dependency entirely.

Reply:
hamcrest doesnt seem to support recursive comparision.